### DISCUSSION TITLE
Why does the Handle method of IHandleRequests<TRequest> return TRequest?

### DISCUSSION THREAD

**Discussion by joe-hamilton-wesd** [BrighterCommand/Brighter#2360] [Q&A] [answered]
Why does the Handle method of IHandleRequests<TRequest> return a request TRequest (command or event) instead of void? Is that a requirement in order for the framework to work? I don't see how that is being utilized.

**Comment by iancooper** [comment, marked as answer]
Sorry @ninkasi99 I missed this. It allows us to chain middleware handlers together:

If you want to deep dive: https://brightercommand.gitbook.io/paramore-brighter-documentation/under-the-hood/howbrighterworks
